packages/block-library/src/text-columns/index.js

```diff
@@ -18,44 +18,18 @@ import {
 } from '@wordpress/block-editor';
 import deprecated from '@wordpress/deprecated';
 
-export const name = 'core/text-columns';
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
 
-export const settings = {
-	// Disable insertion as this block is deprecated and ultimately replaced by the Columns block.
-	supports: {
-		inserter: false,
-	},
+export { metadata };
```

Comment:
I guess the idea later is to have a feature flag to set metadata to undefined in Core and Gutenberg when these are provided by the server right?

Reply:
Yes, exactly. That would be optimal to use dead-code elimination. I'm not quite sure how it would work with deprecations which depend on `attributes` though :)

Example:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/gallery/index.js#L21-L71
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/gallery/index.js#L229
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/gallery/index.js#L289
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/gallery/index.js#L324-L334